runtime/compiler/typst.vim

```diff
@@ -0,0 +1,34 @@
+" Vim compiler file
+" Compiler: typst
+
+if exists('current_compiler')
+    finish
+endif
+let current_compiler = get(g:, 'typst_cmd', 'typst')
+
+let s:save_cpo = &cpo
+set cpo&vim
+
+if exists(":CompilerSet") != 2
+    command -nargs=* CompilerSet setlocal <args>
+endif
+
+" With `--diagnostic-format` we can use the default errorformat
+let s:makeprg = [current_compiler, 'compile',
+              \  '--diagnostic-format', 'short']
+
+if has('patch-7.4.191')
+    call add(s:makeprg, '%:S')
+else
+    call add(s:makeprg, '%')
+endif
```

Comment:
Since we are including this for Vim 9.1, I think we can remove the test for patch-7.4.191 (after all, that's been 10 years).